src/Time/TimeSteppers/Cerk3.cpp

```diff
@@ -0,0 +1,84 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Time/TimeSteppers/Cerk3.hpp"
+
+#include <cmath>
+#include <limits>
+
+#include "Time/TimeStepId.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace TimeSteppers {
+
+size_t Cerk3::order() const { return 3; }
+
+size_t Cerk3::error_estimate_order() const { return 2; }
+
+uint64_t Cerk3::number_of_substeps() const { return 4; }
+
+uint64_t Cerk3::number_of_substeps_for_error() const { return 3; }
```

Comment:
The method requires *fewer* substeps when producing an error estimate?

Reply:
Sorry, I don't follow.  The distinction between these two functions isn't related to dense output.  If the method can provide output with an error estimate in 3 substeps, there's no way it could require 4 for output without an error estimate, because we could just generate the error estimate in 3 and throw it away.

I think this may be related to the other thread on dense output and FSAL, so I'm going to hold off commenting on that one until we've resolved what's going on here.

Reply:
Right, so we need 3 substeps to take the time step and provide an error estimate, but we need 4 for dense/continuous output. Is there a `number_of_substeps_for_dense_output()` function I should've defined?

Reply:
Dense output is always available.  The only distinction between these is whether the error estimate is generated.  And it's tested, so you are apparently successfully providing dense output with 3 substeps.

Reply:
Sorry, I'm not sure what you mean by "dense output is always available". The code explicitly uses `k4`, so needs 4 RHS evaluations with the final one being at the future time step. At that point dense output can be done at full order of accuracy. Am I counting substeps wrong? I really don't know what to make of your reply. What's the remaining question or actionable change you want?